### PR TITLE
Fix network tests when run in frdm-k64f

### LINKED ISSUE #8171: Tests failing with a stacking error on frdm_k64f
The following tests are failing with a stacking error on frdm_k64f:

samples/drivers/watchdog/test
tests/net/utils/net.util
tests/net/icmpv6/net.icmpv6
tests/net/lib/http_header_fields/net.http

A full regression with device testing was run on frdm_k64f on commit 7a18b083ebe4712fbb258d09c0f53a71b7f46678:
```
$ sanitycheck -p frdm_k64f --device-testing --device-serial /dev/ttyACM0 
Cleaning output directory /home/maureen/zephyr/sanity-out
Building testcase defconfigs...
256 tests selected, 55151 tests discarded due to filters
total complete:   27/ 256  10%  failed:    0

frdm_k64f                 samples/mpu/mpu_stack_guard_test/test              FAILED: timeout
	see: sanity-out/frdm_k64f/samples/mpu/mpu_stack_guard_test/test/handler.log

total complete:   46/ 256  17%  failed:    1

frdm_k64f                 samples/mpu/mpu_stack_guard_test/test_stack_guard  FAILED: timeout
	see: sanity-out/frdm_k64f/samples/mpu/mpu_stack_guard_test/test_stack_guard/handler.log

total complete:  112/ 256  43%  failed:    2

frdm_k64f                 tests/net/vlan/net.vlan                            FAILED: failed
	see: sanity-out/frdm_k64f/tests/net/vlan/net.vlan/handler.log

total complete:  126/ 256  49%  failed:    3

frdm_k64f                 samples/drivers/watchdog/test                      FAILED: timeout
	see: sanity-out/frdm_k64f/samples/drivers/watchdog/test/handler.log

total complete:  144/ 256  56%  failed:    4

frdm_k64f                 tests/net/utils/net.util                           FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/utils/net.util/handler.log

total complete:  158/ 256  61%  failed:    5

frdm_k64f                 tests/net/icmpv6/net.icmpv6                        FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/icmpv6/net.icmpv6/handler.log

total complete:  159/ 256  62%  failed:    6

frdm_k64f                 tests/benchmarks/sys_kernel/benchmark.kernel       FAILED: failed
	see: sanity-out/frdm_k64f/tests/benchmarks/sys_kernel/benchmark.kernel/handler.log

total complete:  173/ 256  67%  failed:    7

frdm_k64f                 tests/net/checksum_offload/net.offload             FAILED: failed
	see: sanity-out/frdm_k64f/tests/net/checksum_offload/net.offload/handler.log

total complete:  196/ 256  76%  failed:    8

frdm_k64f                 tests/net/lib/http_header_fields/net.http          FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/lib/http_header_fields/net.http/handler.log

total complete:  207/ 256  80%  failed:    9

frdm_k64f                 tests/net/trickle/net.trickle                      FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/trickle/net.trickle/handler.log

total complete:  208/ 256  81%  failed:   10

frdm_k64f                 tests/kernel/fifo/fifo_api/kernel.fifo.poll        FAILED: timeout
	see: sanity-out/frdm_k64f/tests/kernel/fifo/fifo_api/kernel.fifo.poll/handler.log

total complete:  222/ 256  86%  failed:   11

frdm_k64f                 tests/net/lib/mqtt_packet/net.mqtt.packet          FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/lib/mqtt_packet/net.mqtt.packet/handler.log

total complete:  235/ 256  91%  failed:   12

frdm_k64f                 tests/drivers/adc/adc_simple/peripheral.adc        FAILED: failed
	see: sanity-out/frdm_k64f/tests/drivers/adc/adc_simple/peripheral.adc/handler.log

total complete:  256/ 256  100%  failed:   13
243 of 256 tests passed with 0 warnings in 7090 seconds
```

### LINKED ISSUE #8172: Networking tests failing with an assertion on frdm_k64f
The following networking tests are failing with an assertion on frdm_k64f:

tests/net/vlan/net.vlan
```
***** Booting Zephyr OS v1.12.0-rc2-75-g7a18b08 *****                           
Running test suite net_vlan_test
===================================================================
starting test - test_vlan_setup

    Assertion failed at /home/maureen/zephyr/tests/net/vlan/src/main.c:337: test
Invalid numer of VLANs 8 vs 4

FAIL - test_vlan_setup
```

tests/net/checksum_offload/net.offload
```
***** Booting Zephyr OS v1.12.0-rc2-75-g7a18b08 *****                           
Running test suite net_chksum_offload_test
===================================================================
starting test - eth_setup

    Assertion failed at /home/maureen/zephyr/tests/net/checksum_offload/src/main
Invalid number of interfaces (3 vs 2)

FAIL - eth_setup
```

A full regression with device testing was run on frdm_k64f on commit 7a18b083ebe4712fbb258d09c0f53a71b7f46678:
```
$ sanitycheck -p frdm_k64f --device-testing --device-serial /dev/ttyACM0 
Cleaning output directory /home/maureen/zephyr/sanity-out
Building testcase defconfigs...
256 tests selected, 55151 tests discarded due to filters
total complete:   27/ 256  10%  failed:    0

frdm_k64f                 samples/mpu/mpu_stack_guard_test/test              FAILED: timeout
	see: sanity-out/frdm_k64f/samples/mpu/mpu_stack_guard_test/test/handler.log

total complete:   46/ 256  17%  failed:    1

frdm_k64f                 samples/mpu/mpu_stack_guard_test/test_stack_guard  FAILED: timeout
	see: sanity-out/frdm_k64f/samples/mpu/mpu_stack_guard_test/test_stack_guard/handler.log

total complete:  112/ 256  43%  failed:    2

frdm_k64f                 tests/net/vlan/net.vlan                            FAILED: failed
	see: sanity-out/frdm_k64f/tests/net/vlan/net.vlan/handler.log

total complete:  126/ 256  49%  failed:    3

frdm_k64f                 samples/drivers/watchdog/test                      FAILED: timeout
	see: sanity-out/frdm_k64f/samples/drivers/watchdog/test/handler.log

total complete:  144/ 256  56%  failed:    4

frdm_k64f                 tests/net/utils/net.util                           FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/utils/net.util/handler.log

total complete:  158/ 256  61%  failed:    5

frdm_k64f                 tests/net/icmpv6/net.icmpv6                        FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/icmpv6/net.icmpv6/handler.log

total complete:  159/ 256  62%  failed:    6

frdm_k64f                 tests/benchmarks/sys_kernel/benchmark.kernel       FAILED: failed
	see: sanity-out/frdm_k64f/tests/benchmarks/sys_kernel/benchmark.kernel/handler.log

total complete:  173/ 256  67%  failed:    7

frdm_k64f                 tests/net/checksum_offload/net.offload             FAILED: failed
	see: sanity-out/frdm_k64f/tests/net/checksum_offload/net.offload/handler.log

total complete:  196/ 256  76%  failed:    8

frdm_k64f                 tests/net/lib/http_header_fields/net.http          FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/lib/http_header_fields/net.http/handler.log

total complete:  207/ 256  80%  failed:    9

frdm_k64f                 tests/net/trickle/net.trickle                      FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/trickle/net.trickle/handler.log

total complete:  208/ 256  81%  failed:   10

frdm_k64f                 tests/kernel/fifo/fifo_api/kernel.fifo.poll        FAILED: timeout
	see: sanity-out/frdm_k64f/tests/kernel/fifo/fifo_api/kernel.fifo.poll/handler.log

total complete:  222/ 256  86%  failed:   11

frdm_k64f                 tests/net/lib/mqtt_packet/net.mqtt.packet          FAILED: timeout
	see: sanity-out/frdm_k64f/tests/net/lib/mqtt_packet/net.mqtt.packet/handler.log

total complete:  235/ 256  91%  failed:   12

frdm_k64f                 tests/drivers/adc/adc_simple/peripheral.adc        FAILED: failed
	see: sanity-out/frdm_k64f/tests/drivers/adc/adc_simple/peripheral.adc/handler.log

total complete:  256/ 256  100%  failed:   13
243 of 256 tests passed with 0 warnings in 7090 seconds
```

### PR DESCRIPTION
**Pull request by jukkar**
Fixes #8171 
Fixes #8172

### CODE REVIEW

File: tests/net/checksum_offload/prj.conf [hunk 1]
```diff
@@ -33,3 +33,12 @@ CONFIG_NET_DEBUG_IPV4=n
 CONFIG_NET_DEBUG_NET_PKT=y
 CONFIG_SYS_LOG_NET_LEVEL=4
 CONFIG_NET_SHELL=n
+
+# Disable internal ethernet drivers as the test is self contained
+# and does not need the on board driver to function.
+CONFIG_ETH_NATIVE_POSIX=n
```
Comment (nashif) [inline tests/net/checksum_offload/prj.conf]: We have no way if disabling ethernet completely, for all devices? for example CONFIG_ETH=n?

Comment (jukkar) [reply tests/net/checksum_offload/prj.conf]: Not that I know of. And beside we do not want to disable ethernet as that would disable the test, we just want to disable on board ethernet chip if the test is run on such board that has ethernet port.

Comment (nashif) [reply tests/net/checksum_offload/prj.conf]: ok

**Comment by nashif** [comment]
this one + this PR will lead to 100% test pass on frdm_k64f:

```
--- a/tests/net/lib/dns_packet/prj.conf
+++ b/tests/net/lib/dns_packet/prj.conf
@@ -7,6 +7,6 @@ CONFIG_ENTROPY_GENERATOR=y
 CONFIG_TEST_RANDOM_GENERATOR=y

 CONFIG_DNS_RESOLVER=y
-
 CONFIG_PRINTK=y
 CONFIG_ZTEST=y
+CONFIG_MAIN_STACK_SIZE=1024
```

**Comment by jukkar** [comment]
> this one + this PR will lead to 100% test pass on frdm_k64f:

I did not know the dns_packet test was also failing, I can add that to this pr too.

**Comment by jukkar** [comment]
Fixed dns_packet test stack size.

**Comment by MaureenHelm** [comment]
Leave a comment

> I did not know the dns_packet test was also failing, I can add that to this pr too.

Neither did I. @nashif did you enable the test recently?

The tests I reported as failing now pass with this PR.

**Comment by nashif** [comment]
> Neither did I. @nashif did you enable the test recently?

no, not sure why it was missed in your run

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
sanitycheck --device-testing --device-serial /dev/ttyACM1 -p frdm_k64f -T tests/net/

now completes 40/40

**Pull request merged**